chat.js

```diff
@@ -350,6 +353,38 @@ class CommandContext {
 		}
 		return true;
 	}
+	checkR9k(user, room, message) {
+		let checksum = Chat.md5(toId(message)).substr(0, 16);
```

Comment:
Note that this will have a ~50% collision rate at 4 billion messages.

(16 hex characters = 8 bytes; 50% collision rate at 2^(8 * 8 / 2) = 4 billion)